GVFS/GVFS.Common/ProductUpgrader.cs

```diff
@@ -179,7 +179,9 @@ public virtual bool TrySetupUpgradeApplicationDirectory(out string upgradeApplic
             error = null;
             try
             {
-                this.fileSystem.CopyDirectoryRecursive(currentPath, upgradeApplicationDirectory);
+                HashSet<string> directoriesToExclude = new HashSet<string>();
+                directoriesToExclude.Add(GVFSPlatform.Instance.GetDataRootForGVFS());
+                this.fileSystem.CopyDirectoryRecursive(currentPath, upgradeApplicationDirectory, directoriesToExclude);

```

Comment:
How did this work before?  I am wondering why we now need to specify directories to exclude.

Reply:
It used to copy the contents of `C:\Program Files\GVFS` to `C:\ProgramData\GVFS.Upgrade\Tools`  directory and run the upgrade tool from `C:\ProgramData\GVFS.Upgrade\Tools`. 

But now we copy `C:\Program Files\GVFS` to inside of `C:\Program Files\GVFS\ProgramData\GVFS.Upgrade\Tools` directory. This causes a cycle (at some point we start copying `C:\Program Files\GVFS\ProgramData\GVFS.Upgrade` and its contents into `C:\Program Files\GVFS\ProgramData\GVFS.Upgrade\Tools`). The exclusion is added to avoid this loop.

Reply:
Ok thanks!  Could you add a comment to the code here explaining why this directory is being excluded?

Reply:
[added](https://github.com/microsoft/VFSForGit/pull/1597/commits/bccac470eca503294beb8e58d9780991f9590af0)